redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java

```diff
@@ -138,23 +149,37 @@ private SecurityGroup buildSecurityGroup(SecurityGroupV4Request source) {
         return securityGroup;
     }
 
-    // private Json getTags(String owner, String cloudPlatform) {
-    //     try {
-    //         return new Json(new StackTags(new HashMap<>(), new HashMap<>(), getDefaultTags(owner, cloudPlatform)));
-    //     } catch (Exception ignored) {
-    //         throw new BadRequestException("Failed to convert dynamic tags.");
-    //     }
-    // }
-
-    // private Map<String, String> getDefaultTags(String owner, String cloudPlatform) {
-    //     Map<String, String> result = new HashMap<>();
-    //     try {
-    //         result.putAll(costTaggingService.prepareDefaultTags(owner, result, cloudPlatform));
-    //     } catch (Exception e) {
-    //         LOGGER.debug("Exception during reading default tags.", e);
-    //     }
-    //     return result;
-    // }
+    // compare to freeipa CostTaggingService
+
+    private Json getTags(Crn ownerCrn, String cloudPlatform) {
+        // freeipa currently uses account ID for username / owner
+        String user = ownerCrn.getResource().toString();
+
+        Map<String, String> defaultTags = new HashMap<>();
+        defaultTags.put(safeTagString(CB_USER_NAME.key(), cloudPlatform), safeTagString(user, cloudPlatform));
+        defaultTags.put(safeTagString(CB_VERSION.key(), cloudPlatform), safeTagString(version, cloudPlatform));
+        defaultTags.put(safeTagString(OWNER.key(), cloudPlatform), safeTagString(user, cloudPlatform));
+        defaultTags.put(safeTagString(CB_CREATION_TIMESTAMP.key(), cloudPlatform),
+            safeTagString(String.valueOf(clock.getCurrentInstant().getEpochSecond()), cloudPlatform));
+
+        return new Json(new StackTags(new HashMap<>(), new HashMap<>(), defaultTags));
+    }
+
+    private static String safeTagString(String value, String platform) {
+        String valueAfterCheck = Strings.isNullOrEmpty(value) ? "unknown" : value;
+        return CloudConstants.GCP.equals(platform)
+                ? valueAfterCheck.split("@")[0].toLowerCase().replaceAll("[^\\w]", "-") : valueAfterCheck;
+    }
+
+    // Sorry, MissingResourceNameGenerator seems like overkill
```

Comment:
?

Reply:
This is a reference to other similar converters which call on a separate, injected name generator objects to make names for resources, but only for certain types of known resources (see ApiResourceType). Since this converter doesn't work like others before it, I wanted to note that. I'll make the comment clearer.